### PR TITLE
Feature/apply prop and state destructuring

### PR DESCRIPTION
**Pull request by Definitely-Not-Vlad**
Follow up PR for prop type and default prop convention.

- destructure `this.props` and `this.state` wherever necessary
- replace weird `const { props } = this;` usage with normal prop destructuring
- removed all possible uses of `delete`, replaced them with `const { deletedProperty, ...otherProperties } = object;`

### CODE REVIEW

File: components/DropDownMenu/DropDownModal.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ class DropDownModal extends PureComponent {
   }
 
   emitOnOptionSelectedEvent(option) {
-    if (this.props.onOptionSelected) {
-      this.props.onOptionSelected(option);
-    }
+    const { onOptionSelected } = this.props;
+
+    onOptionSelected && onOptionSelected(option);
```
Comment (tomislav-arambasic) [inline components/DropDownMenu/DropDownModal.js:71]: I'm not sure if we have a convention on this, but we should.

I'm fine with both,`if` or ternary conditional callbacks, but we should agree on one of those.

I think I'd prefer `if` more, kinda easier to read

File: components/ImageGallery/ImageGalleryBase.js [hunk 2]
```diff
@@ -29,11 +29,15 @@ export class ImageGalleryBase extends PureComponent {
 
     autoBindReact(this);
 
+    const { selectedIndex } = props;
+
     this.state = {
-      selectedIndex: this.props.selectedIndex || 0,
+      selectedIndex,
```
Comment (tomislav-arambasic) [inline components/ImageGallery/ImageGalleryBase.js:35]: You sure we don't need fallback `0` here?
If we do, please add it to `defaultProps`

File: components/ScrollView/ScrollView.js [hunk 3]
```diff
@@ -77,19 +77,17 @@ class ScrollView extends PureComponent {
...
+    const { contentContainerStyle, ...otherStyle } = style;
 
     return (
       <Animated.ScrollView
         ref={this.setWrappedInstance}
         contentContainerStyle={this.addIphoneXPadding(contentContainerStyle)}
-        {...animationDriver.scrollViewProps}
-        {..._.omit(props, 'onScroll')}
+        {...scrollViewProps}
+        {..._.omit(otherProps, 'onScroll')}
```
Comment (tomislav-arambasic) [inline components/ScrollView/ScrollView.js:89]: Bruh, we need to memoize this asap (ik it wasn't part of this PR, just sayin')

File: components/View.js [hunk 4]
```diff
@@ -7,25 +7,23 @@ import { LinearGradient } from './LinearGradient';
...
 class View extends PureComponent {
   render() {
-    const style = { ...this.props.style };
-    let gradient = null;
+    const { children, style, ...otherProps } = this.props;
+    const { backgroundGradient, ...viewStyle } = style;
 
-    if (style.backgroundGradient) {
+    let gradient = null;
+    if (backgroundGradient) {
```
Comment (tomislav-arambasic) [inline components/View.js:14]: Empty line above please

**Review by tomislav-arambasic — CHANGES_REQUESTED** [review]
Thanks for doing this.
Requesting changes only because of fallback 0 value, I'm not sure if not having it could break things

File: components/ImageGallery/ImageGalleryBase.js (hunk 2 above)
Comment (Definitely-Not-Vlad) [reply components/ImageGallery/ImageGalleryBase.js:35]: Already done in previous PR in chain where defaultProps are defined.

File: components/DropDownMenu/DropDownModal.js (hunk 1 above)
Comment (Definitely-Not-Vlad) [reply components/DropDownMenu/DropDownModal.js:71]: This is not ternary, this is just how `&&` works, if both are true, it returns the right hand side, so you can expect it to execute the function passed to it.

I prefer this, obviously, but we can put it to vote in a dev-weekly, I'll add it to the agenda as a tiny point.

Until then, I'll use `if (x) { x() }`

Comment (sstimac) [reply components/DropDownMenu/DropDownModal.js:71]: safest would be if (_.isFunction(callback)), but we could also set this to requiredProp and resolve the issue that way

**Review by sstimac — APPROVED** [review]
very nice. thanks for doing this

**Review by tomislav-arambasic — APPROVED** [review]

**Pull request merged**